sentry-core/src/main/java/io/sentry/core/cache/CacheStrategy.java

```diff
@@ -0,0 +1,97 @@
+package io.sentry.core.cache;
+
+import static io.sentry.core.SentryLevel.ERROR;
+
+import io.sentry.core.ISerializer;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import io.sentry.core.util.Objects;
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import org.jetbrains.annotations.NotNull;
+
+abstract class CacheStrategy {
+
+  @SuppressWarnings("CharsetObjectCanBeUsed")
+  protected static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  protected final @NotNull SentryOptions options;
+  protected final @NotNull ISerializer serializer;
+  protected @NotNull File directory;
+  protected int maxSize;
+
+  CacheStrategy(
+      final @NotNull SentryOptions options,
+      final @NotNull String directoryPath,
+      final int maxSize) {
+    this.options = Objects.requireNonNull(options, "SentryOptions is required.");
+    this.serializer = options.getSerializer();
+
+    Objects.requireNonNull(directoryPath, "Directory is required.");
+    this.directory = new File(directoryPath);
+
+    this.maxSize = maxSize;
+  }
+
+  /**
+   * Check if a Dir. is valid and have write and read permission
+   *
+   * @return true if valid and has permissions or false otherwise
+   */
+  protected boolean isDirectoryValid() {
+    if (!directory.isDirectory() || !directory.canWrite() || !directory.canRead()) {
+      options
+          .getLogger()
+          .log(
+              ERROR,
+              "The directory for caching files is inaccessible.: %s",
+              directory.getAbsolutePath());
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Sort files from oldest to the newest using the lastModified method
+   *
+   * @param files the Files
+   */
+  void sortFilesOldestToNewest(@NotNull File[] files) {
+    // just sort it if more than 1 file
+    if (files.length > 1) {
+      // sort by the oldest to the newest
```

Comment:
`l`: I think we could skip this comment here as the method name already tells me what it does.

Reply:
true, I renamed the method, now the comment doesn't make sense, will change it